runtime/interpreter/interpreter_statement.go

```diff
@@ -318,6 +329,11 @@ func (interpreter *Interpreter) VisitForStatement(statement *ast.ForStatement) a
 		case functionReturn:
 			return result
 		}
+
+		if indexVariable != nil {
+			index++
+			indexValue.BigInt.SetInt64(index)
```

Comment:
Is it safe to change the existing value, or should we assign a new value? I feel it might also mutate the usages of previous indexes as well.

like for eg: what if a user stores the indexes in an array, and prints them later?

Reply:
Ah, that's a good point. I had suggested this above to avoid allocation, but I think you are right

Reply:
I added a test for this, and there are is indeed an error in this case. Nice catch

Reply:
@dsainati1 Nice, good idea to add a test! Feel free to revert to your original code, that worked and I think was safe 👍